lib/fog/compute/google/models/instance_group.rb

```diff
@@ -27,9 +27,39 @@ def destroy(_async = true)
           data = service.delete_instance_group(name, zone_name)
         end
 
+        def add_instances(instances)
+          requires :identity, :zone
+
+          service.add_instance_group_instances(
+            identity, zone_name, format_instance_list(instances)
+          )
+        end
+
+        def remove_instances(instances)
+          requires :identity, :zone
+
+          service.remove_instance_group_instances(
+            identity, zone_name, format_instance_list(instances)
+          )
+        end
+
+        def list_instances
+          requires :identity, :zone
+
+          data = service.list_instance_group_instances(identity, zone_name).body
+          data["items"]
+        end
+
         def zone_name
           zone.nil? ? nil : zone.split("/")[-1]
         end
+
+        private
+
+        def format_instance_list(instance_list)
```

Comment:
IMHO this is right. In the past we had some requests that were overloaded with such kind of logic and it was a PITA because test it is hard since you gotta mock the actual request somehow.